controllers/authController.js

```diff
@@ -64,3 +79,73 @@ exports.loginUser = (req, res) => {
         return res.status(200).json({message: "Successfully logged in", token})
   })
 }
+
+//This function is responsible for sending email to a particular user with the token for password reset
+exports.forgotPassword = (req, res) => {
+  User.findOne({email: req.body.email} , (err, foundUser) => {
+    if (err) {
+      return res.status(400).json({err})
+    }
+    if (!foundUser) {
+      return res.status(401).json({message: "User with this email does not exist"})
+    }
+    let token = await Token.findOne({ userId: user._id});
+    if (!token) {return res.status(400).send("Invalid user");}
+    token = await new Token({
+      userId: user._id,
+      email: req.body.email,
+      token: bcrypt.randomBytes(32).toString("hex"),
+      createdAt: Date.now(),
+    }).save();
+
+    const link = `${process.env.BASE_URL}/forgotPassword/${user._id}/${token.token}`;
+    User.link = link;
+    User.save();
+    await sendEmail(user.email, "Password reset",link);
+  })
+}
+
+//
+exports.resetPassword = (req, res) => {
+  const {link, newPass} = req.body;
```

Comment:
I think what you want to do is to let the user send the token (the one you have on line 101 as `token.token`), then `Token.findOne({token:token})`